src/poli/objective_repository/rosetta_energy/register.py

```diff
@@ -0,0 +1,329 @@
+        Scaling factor applied when computing DDGs (DDG=deltaREU/conversion_factor). Only applies to DDG conversion.
+        Default is '2.9' following:
+            Park, Hahnbeom, et al.
+            "Simultaneous optimization of biomolecular energy functions on features from small molecules and macromolecules."
+            Journal of chemical theory and computation 12.12 (2016): 6201-6212.
+            DOI: https://doi.org/10.1021/acs.jctc.6b00819
+    clean : bool, optional
+        Flag whether to apply PDB cleaning routine prior to pose loading of reference.
+        Default is True.
+    relax: bool, optional
+        Flag whether to apply relaxation protocol. Default enabled.
+        NOTE: disable if fast compute required
+    pack: bool, optional
+        Flag whether to apply packing protocol. Default enabled.
+        NOTE: disable if fast compute required
+    cycle: int, optional
+        Number of relaxation cycles applied.
+    constraint_weight: int, optional
+        Constraint on scoring function for atom-pair, angles, coordinates.
+
+    Methods
+    -------
+    _black_box(x, context=None)
+        The main black box method that performs the computation, i.e.
+        it computes the stability of the mutant(s) in x.
+    get_black_box_info()
+        Returns BlackBox info object.
+
+    Raises
+    ------
+    AssertionError
+        If wildtype_pdb_path is not provided.
+        If the provided energy units are not valid.
+    NotImplementedError
+        If energy function does not exist.
+
+    Notes
+    -----
+    - The wildtype_pdb_path is a single Path object.
+    """
+
+    def __init__(
+        self,
+        wildtype_pdb_path: Path | List[Path],
+        score_function: str = "default",
+        seed: int = 0,
+        unit: str = "DDG",
+        conversion_factor: float = 2.9,
+        clean: bool = True,
+        relax: bool = True,
+        pack: bool = True,
+        cycle: int = 3,
+        constraint_weight: float = 5.0,
+        n_threads: int = 4,
+        batch_size: int = None,
+        parallelize: bool = False,
+        num_workers: int = None,
+        evaluation_budget: int = float("inf"),
+        force_isolation: bool = False,
+    ):
+        super().__init__(
+            batch_size=batch_size,
+            parallelize=parallelize,
+            num_workers=num_workers,
+            evaluation_budget=evaluation_budget,
+        )
+        assert wildtype_pdb_path is not None
+        self.force_isolation = force_isolation
+        self.wildtype_pdb_path = wildtype_pdb_path
+        self.score_function = score_function
+        self.seed = seed
+        self.unit = unit
+        self.conversion_factor = conversion_factor
+        self.clean = clean
+        self.relax = relax
+        self.pack = pack
+        self.cycle = cycle
+        self.constraint_weight = constraint_weight
+        self.n_threads = n_threads
+
+        try:
+
+            inner_function = get_inner_function(
```

Comment:
If you check the implementation of get inner function, it already contains the logic of the try-except! :)

Reply:
Noted. Thanks for the hint! `try ... except ...` taken out.